extensions/git/src/git.ts

```diff
@@ -1536,6 +1536,25 @@ export class Repository {
 		await this.exec(args);
 	}
 
+	async fastForwardBranch(options: { remote: string; ref: string; readonly cancellationToken?: CancellationToken }): Promise<void> {
+		const args = ['fetch', options.remote, `${options.ref}:${options.ref}`];
```

Comment:
What if the upstream ref has a different name than the local ref?

Reply:
Great point! Fixed!